test/unit/mt_membar.c

```diff
@@ -0,0 +1,189 @@
+#include <stdio.h>
+#include <assert.h>
+#include <pthread.h>
+#include <shmem.h>
+
+/* For systems without the PThread barrier API (e.g. MacOS) */
+#include "pthread_barrier.h"
+
+#define T 2
+#define ITER 100
+
+#ifndef MAX
+#define MAX(A,B)   (((A)>(B)) ? (A) : (B))
+#endif
+
+int shared_dest_1 = 0, shared_dest_2 = 0, result = 0;
+int me, npes, errors = 0, sum_error = 0;
+long lock = 0;
+
+pthread_barrier_t fencebar;
+
+long pSync[_SHMEM_REDUCE_SYNC_SIZE];
+int pWrk[MAX(1, _SHMEM_REDUCE_MIN_WRKDATA_SIZE)];
```

Comment:
The `_SHMEM*` constants are deprecated, please use the `SHMEM*` name for these constants.